### PR TITLE
bandit v3

### PR DESCRIPTION
**Pull request by YufanWangYuki**

### CODE REVIEW

File: src/bbb/tasks/bandit.py [hunk 1]
```diff
@@ -117,51 +121,54 @@ def loss_step(self, X: torch.Tensor, y: torch.Tensor):
...
 class Greedy(MushroomBandit):
-    def __init__(self, epsilon=0, lr=2e-5, **kwargs):
-        super().__init__(**kwargs)
+    def __init__(self, epsilon=0):
+        super().__init__()
         self.n_weight_sampling = 1
         self.epsilon = epsilon
-        self.net = RegressionDNN(params=DNN_REGRESSION_PARAMETERS).to(DEVICE)
-        self.optimizer = optim.SGD(self.net.parameters(), lr=lr)
+        self.net = DNN(params=DNN_REGRESSION_PARAMETERS).to(DEVICE)
```
Comment (thatscotdatasci) [inline src/bbb/tasks/bandit.py:136]: This needs to be RegressionDNN now :)

File: src/bbb/tasks/bandit.py [hunk 2]
```diff
@@ -170,39 +177,42 @@ def loss_step(self, x, y, batch_id):
...
+    def __init__(self):
+        super().__init__()
+        self.n_weight_sampling = 2
         self.net = BanditBNN(params=BNN_REGRESSION_PARAMETERS).to(DEVICE)
-        self.optimizer = optim.SGD(self.net.model.parameters(), lr=lr)
         
     def loss_step(self, x, y, batch_id):
         beta = 2 ** (64 - (batch_id + 1)) / (2 ** 64 - 1)
         beta = torch.Tensor((beta,)).to(DEVICE)
-        self.net.model.train()
```
Comment (thatscotdatasci) [inline src/bbb/tasks/bandit.py:182]: `self.net.train()` needs to be present

File: src/bbb/tasks/bandit.py [hunk 3]
```diff
@@ -117,51 +121,54 @@ def loss_step(self, X: torch.Tensor, y: torch.Tensor):
...
+        super().__init__()
         self.n_weight_sampling = 1
         self.epsilon = epsilon
-        self.net = RegressionDNN(params=DNN_REGRESSION_PARAMETERS).to(DEVICE)
-        self.optimizer = optim.SGD(self.net.parameters(), lr=lr)
+        self.net = DNN(params=DNN_REGRESSION_PARAMETERS).to(DEVICE)
         self.criterion = torch.nn.MSELoss()
         
     def loss_step(self, x, y, batch_id):
-
```
Comment (thatscotdatasci) [inline src/bbb/tasks/bandit.py:137]: `self.net.train()` needs to be present

File: src/bbb/tasks/bandit.py [hunk 4]
```diff
@@ -117,51 +121,54 @@ def loss_step(self, X: torch.Tensor, y: torch.Tensor):
...
         self.net.zero_grad()
         preds = self.net.forward(x)
         loss = self.criterion(preds, y)
         loss.backward()
-        self.optimizer.step()
+        self.net.optimizer.step()
         return loss
 
 BNN_REGRESSION_PARAMETERS = Parameters(
     name = "BBB_regression",
```
Comment (thatscotdatasci) [inline src/bbb/tasks/bandit.py:148]: This should be `BBB_rl`; also need to change `DNN_regression` to `DNN_rl`

Comment (YufanWangYuki) [reply src/bbb/tasks/bandit.py:148]: What do you mean by this one? The name of the parameters I guess?

**Pull request merged**